
mobile_base_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel: 'left_wheel_joint'
  right_wheel: 'right_wheel_joint'
  publish_rate: 50.0               # default: 50
  pose_covariance_diagonal : [0.01, 0.01, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.01, 0.01, 1000000.0, 1000000.0, 1000000.0, 1000.0]

  # Wheel separation and diameter. These are both optional.
  # diff_drive_controller will attempt to read either one or both from the
  # URDF if not specified as a parameter
  wheel_separation : 0.3284
  wheel_radius : 0.1

  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 0.25

  enable_odom_tf: true

  # Base frame_id
  base_frame_id: base_link

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 2.0  # m/s
      min_velocity           : -2.0 # m/s
      has_acceleration_limits: true
      max_acceleration       : 1.0  # m/s^2
      min_acceleration       : -1.0 # m/s^2
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 3.0  # rad/s
      has_acceleration_limits: true
      max_acceleration       : 1.5  # rad/s^2
      

robot_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

elevator_controller:
  type: position_controllers/JointPositionController
  joint: base_elevator_joint
  
  
torso_trajectory_controller:
  type: position_controllers/JointTrajectoryController
  joints:
     - torso_joint
  constraints:
      goal_time: 4.0
      stopped_velocity_tolerance: 0.05
      base_elevator_joint: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 5.0
  state_publish_rate:  50
  action_monitor_rate: 50
     
pan_tilt_controller:
    type: position_controllers/JointGroupPositionController
    joints: ['head_pan_joint', 'head_tilt_joint']
    publish_rate: 20
  
arm_position_controller:
    type: position_controllers/JointGroupPositionController
    joints: ['rotation1_joint', 'shoulder1_joint', 'shoulder2_joint', 'rotation2_joint', 'shoulder3_joint', 'wrist_joint']
    publish_rate: 20


arm_trajectory_controller:
  type: position_velocity_controllers/JointTrajectoryController
  joints:
     - rotation1_joint
     - shoulder1_joint 
     - shoulder2_joint 
     - rotation2_joint 
     - shoulder3_joint 
     - wrist_joint 
  constraints:
      goal_time: 4.0
      stopped_velocity_tolerance: 0.05
      rotation1_joint: {trajectory: 0.1, goal: 0.1}
      shoulder1_joint: {trajectory: 0.1, goal: 0.1}
      shoulder2_joint: {trajectory: 0.1, goal: 0.1}
      rotation2_joint: {trajectory: 0.1, goal: 0.1}
      shoulder3_joint: {trajectory: 0.1, goal: 0.1}
      wrist_joint: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 5.0
  state_publish_rate:  50
  action_monitor_rate: 50


gripper_controller:
 type: position_controllers/GripperActionControllerTwo
 leftJoint: 'left_finger_joint'
 rightJoint: 'right_finger_joint'
 goal_tolerance: 0.01
 stalled_velocity_threshold: 0.005
 stall_timeout: 5.0
